src/__tests__/verifyKey.ts

```diff
@@ -1,170 +1,167 @@
-				signedRequest.timestamp,
-				validKeyPair.publicKey,
-			),
-		).toBe(true);
+			validKeyPair.privateKey,
+		);
+		const isValid = await verifyKey(
+			signedRequest.body,
+			signedRequest.signature,
+			signedRequest.timestamp,
+			validKeyPair.publicKey,
+		);
+		expect(isValid).toBe(true);
 	});
 
-	it('invalid key', () => {
+	it('invalid key', async () => {
 		// Sign a request with a different private key and verify with the valid public key
-		const signedRequest = signRequestWithKeyPair(
+		const signedRequest = await signRequestWithKeyPair(
 			pingRequestBody,
-			invalidKeyPair.secretKey,
-		);
-		expect(
-			verifyKey(
-				signedRequest.body,
-				signedRequest.signature,
-				signedRequest.timestamp,
-				validKeyPair.publicKey,
-			),
-		).toBe(false);
+			invalidKeyPair.privateKey,
+		);
+		const isValid = await verifyKey(
+			signedRequest.body,
+			signedRequest.signature,
+			signedRequest.timestamp,
+			validKeyPair.publicKey,
+		);
+		expect(isValid).toBe(false);
 	});
 
-	it('invalid body', () => {
+	it('invalid body', async () => {
 		// Sign a valid request and verify with an invalid body
-		const signedRequest = signRequestWithKeyPair(
+		const signedRequest = await signRequestWithKeyPair(
 			pingRequestBody,
-			validKeyPair.secretKey,
-		);
-		expect(
-			verifyKey(
-				'example invalid body',
-				signedRequest.signature,
-				signedRequest.timestamp,
-				validKeyPair.publicKey,
-			),
-		).toBe(false);
+			validKeyPair.privateKey,
+		);
+		const isValid = await verifyKey(
+			'example invalid body',
+			signedRequest.signature,
+			signedRequest.timestamp,
+			validKeyPair.publicKey,
+		);
+		expect(isValid).toBe(false);
 	});
 
-	it('invalid signature', () => {
+	it('invalid signature', async () => {
 		// Sign a valid request and verify with an invalid signature
-		const signedRequest = signRequestWithKeyPair(
+		const signedRequest = await signRequestWithKeyPair(
 			pingRequestBody,
-			validKeyPair.secretKey,
-		);
-		expect(
-			verifyKey(
-				signedRequest.body,
-				'example invalid signature',
-				signedRequest.timestamp,
-				validKeyPair.publicKey,
-			),
-		).toBe(false);
+			validKeyPair.privateKey,
+		);
+		const isValid = await verifyKey(
+			signedRequest.body,
+			'example invalid signature',
+			signedRequest.timestamp,
+			validKeyPair.publicKey,
+		);
+		expect(isValid).toBe(false);
 	});
 
-	it('invalid timestamp', () => {
+	it('invalid timestamp', async () => {
 		// Sign a valid request and verify with an invalid timestamp
-		const signedRequest = signRequestWithKeyPair(
+		const signedRequest = await signRequestWithKeyPair(
 			pingRequestBody,
-			validKeyPair.secretKey,
-		);
-		expect(
-			verifyKey(
-				'example invalid body',
-				signedRequest.signature,
-				String(Math.round(new Date().getTime() / 1000) - 10000),
-				validKeyPair.publicKey,
-			),
-		).toBe(false);
+			validKeyPair.privateKey,
+		);
+		const isValid = await verifyKey(
+			'example invalid body',
```

Comment:
I somehow managed to mess up this unit test back in January of 2021 and it has gone unnoticed until now. This should be `signedRequest.body` instead of the invalid body, since we should only be supplying an invalid timestamp for this one.

Reply:
hey, thanks!